java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java

```diff
@@ -240,29 +240,52 @@ public void save(Dataset<Row> featureData, Map<String, String> writeOptions)
   }
 
   public void insert(Dataset<Row> featureData) throws IOException, FeatureStoreException {
-    insert(featureData, null, false);
+    if (this.timeTravelFormat == TimeTravelFormat.HUDI)  {
+      insert(featureData, null, false, HudiOperationType.UPSERT, null);
+    } else {
+      insert(featureData, null, false);
+    }
```

Comment:
Don't organize the code like this. There is no point of repeating the same if for every  `insert` overwrite method. 

Just do it once in this `insert` - https://github.com/logicalclocks/feature-store-api/blob/7c704e06ac1ede40a01895d6b502f449fd028861/java/src/main/java/com/logicalclocks/hsfs/FeatureGroup.java#L292 

If the operation is `null` and the type of the feature group is `Hudi` then set the operation to upsert.